### PR TITLE
fix lovefield table

### PR DESCRIPTION
**Pull request by Brooooooklyn**
Please fill in this template.

- [x] Prefer to make your PR against the `types-2.0` branch.
- [ ] Test the change in your own code.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `npm run lint -- package-name` if a `tslint.json` is present.

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Run `tsc` without errors.
- [ ] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header. Base these on the README, *not* on an existing project.

If changing an existing definition:
- [ ] Provide a URL to  documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.

### CODE REVIEW

File: lovefield/lovefield.d.ts [hunk 1]
```diff
@@ -158,12 +158,14 @@ declare namespace lf {
       getNormalizedName(): string
     }
 
-    export interface Table {
+    interface ITable {
```
Comment (freshp86) [inline lovefield/lovefield.d.ts]: Can you explain what error is this fixing? There are several references to `schema.Table` throughout this file, and no reference to `ITable`.

File: lovefield/lovefield.d.ts [hunk 2]
```diff
@@ -158,12 +158,14 @@ declare namespace lf {
...
       as(name: string): Table
       createRow(value: Object): Row
       getName(): string
     }
 
+    export type Table = ITable & { [index: string]: Column }
```
Comment (freshp86) [inline lovefield/lovefield.d.ts]: Don't fully understand this type here. Can you explain what does `{ [index: string]: Column }` mean? An example compiler error and subsequent success with this change would be helpful.

Comment (freshp86) [inline lovefield/lovefield.d.ts]: cc @arthurhsu

Comment (Brooooooklyn) [inline lovefield/lovefield.d.ts]:
```ts
const table = db.getSchema().table(tableName)
const foo = table['key']
```
typeof foo should be Column

**Comment by sandersn** [comment]
@freshp86 @arthurhsu Is this good to go? @Brooooooklyn could you consider adding a test for the newly-added index signature, basically just pasting in the example you gave above?

**Comment by Brooooooklyn** [comment]
I've updated the test to cover this change.

**Comment by freshp86** [comment]
The fix looks fine to me. I should probably replicate that in Lovefield's own definitions [here](https://github.com/google/lovefield/blob/master/dist/lovefield.d.ts). Could you wait for @arthuhsu to also take a look before merging this though?

**Comment by arthurhsu** [comment]
LGTM

**Comment by sandersn** [comment]
@freshp86 if lovefield bundles its own definitions then it should be removed in types-2.0 branch of DefinitelyTyped and added to notNeededPackages.json. Is that the case? I'll open an issue to track that if so.

**Pull request merged**